doc/neps/roadmap.rst

```diff
@@ -0,0 +1,114 @@
+=============
+NumPy Roadmap
+=============
+
+This is a live snapshot of tasks and features we will be investing resources
+in. It may be used to encourage and inspire developers and to search for
+funding.
+
+Interoperability protocols & duck typing
+========================================
+
+`__array_function__`
+--------------------
+See `NEP 18`_ and a sample implementation_
+
+Array Duck-Typing
+-----------------
+`NEP 22`_    `np.asduckarray()`
+
+
+Mixins like `NDArrayOperatorsMixin`:
+
+- for mutable arrays
+- for reduction methods implemented as ufuncs
+
+Better dtypes
+-------------
+
+- Easier custom dtypes
+  - Simplify and/or wrap the current C-API
+  - More consistent support for dtype metadata
+  - Support for writing a dtype in Python
+- New string dtype(s):
+  - Fixed width encoded strings (utf8, latin1, ...)
```

Comment:
What does fixed width mean for utf8?

Reply:
Fixed number of bytes per string. We already have Unicode strings with a fixed number of characters.

Reply:
"fixed width" in the sense of not doing NULL byte padding is important as well IMO!

Reply:
Is there a recommendation for change here?  You are also very welcome to push changes up directly into my branch.

Reply:
Maybe "fixed width storage"?